CHANGELOG.md

```diff
@@ -3,6 +3,10 @@
 * Add `BoolList` class for space-efficient lists of boolean values.
 * Use a stable sort algorithm in the `IterableExtension.sortedBy` method.
 * Add `min`, `max`, `minOrNull` and `maxOrNull` getters to `IterableDoubleExtension`, `IterableNumberExtension` and `IterableIntegerExtension`
+* Change `UnorderedIterableEquality` and `SetEquality` to implement `Equality`
+  with a non-nullable generic. This allows assignment to `Equality` with a
+  non-nullable generic.
```

Comment:
This is a breaking change, you can no longer use it as equality for `Set<E>?`s.

This is the change I was talking about doing yesterday, but I think it needs to be a major version increment.
(Also, I'd add `class NullableEquality<E extends Object> implements Equality<E?>` which extends equality to from a type to the nullable type by equalting `null` only with itself).